context/http_test.go

```diff
@@ -110,6 +110,13 @@ func (trw *testResponseWriter) Header() http.Header {
 	return trw.header
 }
 
+// CloseNotify is only here to make the testResponseWriter implement the
+// http.CloseNotifier interface, which WithResponseWriter expects to be
+// implemented.
+func (trw *testResponseWriter) CloseNotify() <-chan bool {
+	return nil
```

Comment:
This might cause odd behavior during testing. Operations on a nil channel always proceed, so anything consuming this will get false values coming down the channel.